static/app/components/modals/sudoModal.tsx

```diff
@@ -44,76 +54,94 @@ type Props = WithRouterProps &
     retryRequest?: () => Promise<any>;
   };
 
-type State = {
-  authenticators: Array<Authenticator>;
-  busy: boolean;
-  error: boolean;
-  errorType: string;
-  showAccessForms: boolean;
-  superuserAccessCategory: string;
-  superuserReason: string;
-};
-
-class SudoModal extends Component<Props, State> {
-  static defaultProps: DefaultProps = {
-    closeButton: true,
-  };
-
-  state: State = {
-    authenticators: [],
-    busy: false,
+function SudoModal({
+  closeModal,
+  isSuperuser,
+  location,
+  needsReload,
+  router,
+  retryRequest,
+  api,
+  Header,
+  Body,
```

Comment:
could any of these props become imports or hooks? for example we could replace `withSentryRouter` with `const router = useRouter()` and define it in the component itself.